geo/src/algorithm/intersects/line_string.rs

```diff
@@ -86,6 +91,21 @@ mod test {
     use super::*;
     use crate::wkt;
 
+    #[test]
+    fn test_degenerate_linestring() {
+        let pt: Point<f64> = wkt! {POINT(1 1)}.convert();
+        let poly: Polygon<f64> = wkt! {RECT(0 0, 1 1)}.to_polygon().convert();
+        let ls1: LineString<f64> = wkt! {LINESTRING(1 1)}.convert();
+        let ls2: LineString<f64> = wkt! {LINESTRING(1 1, 1 1)}.convert();
+
+        assert_eq!(ls1.intersects(&pt), ls1.relate(&pt).is_intersects());
+        assert_eq!(ls2.intersects(&pt), ls2.relate(&pt).is_intersects());
+        assert!(ls1.intersects(&poly));
+        // TODO: uncomment when relate is fixed, this currently panics
+        // assert_eq!(ls1.intersects(&poly), ls1.relate(&poly).is_intersects());
```

Comment:
Looks like you opened #1482 to track this. Thank you!

Good find. This seems like a bug in CoordinatePosition. I'm taking a look now.